### ISSUE TITLE
Port tests from Filecoin's HAMT

### ISSUE THREAD

**Issue by Stebalien** [ipld/go-ipld-adl-hamt#20]
Filecoin's HAMT has a lot of important tests around deterministic serialization, limits, etc. We should port as many tests as possible.

**Comment by mvdan** [comment]
Right, though this should come after https://github.com/ipld/go-ipld-adl-hamt/issues/17, because this HAMT as-is is not finished.
